local/src/client/verifier-client.cc

```diff
@@ -276,6 +276,21 @@ ClientReplayFileHandler::ssn_open(YAML::Node const &node)
         _ssn->is_h3 = true;
       }
     }
+    // parse the proxy protocol node
+    auto const pp_node = parse_for_protocol_node(protocol_sequence_node, YAML_SSN_PROTOCOL_PP_NAME);
+    if (!pp_node.is_ok()) {
+      errata.note(std::move(pp_node.errata()));
+      return errata;
+    }
+    if (pp_node.result().IsDefined()) {
+      errata.note(S_DIAG, "Enabling proxy protocol for this session.");
+      // set proxy protocol header version if specified
+      auto const &pp_version_node = pp_node.result()[YAML_SSN_PROTOCOL_VERSION];
+      // for unspecified or unknown versions, default to v1
+      _ssn->pp_version = (pp_version_node && pp_version_node.Scalar() == "2") ?
+                             ProxyProtocolVersion::V2 :
+                             ProxyProtocolVersion::V1;
```

Comment:
Let's check for 1 or 2 and fail hard with an error for any other value.

Reply:
I updated the version checking logic.